deployment/stacks/ui.ts

```diff
@@ -0,0 +1,239 @@
+        },
+        contentSecurityPolicy: {
+          contentSecurityPolicy:
+            "default-src 'self'; img-src 'self' data: https://www.google-analytics.com; script-src 'self' https://www.google-analytics.com https://ssl.google-analytics.com https://www.googletagmanager.com tags.tiqcdn.com tags.tiqcdn.cn tags-eu.tiqcdn.com tealium-tags.cms.gov dap.digitalgov.gov https://*.adoberesources.net 'unsafe-inline'; style-src 'self' maxcdn.bootstrapcdn.com fonts.googleapis.com 'unsafe-inline'; font-src 'self' maxcdn.bootstrapcdn.com fonts.gstatic.com; connect-src https://*.amazonaws.com/ https://*.amazoncognito.com https://www.google-analytics.com https://*.launchdarkly.us https://adobe-ep.cms.gov https://adobedc.demdex.net; frame-ancestors 'none'; object-src 'none'",
+          override: true,
+        },
+      },
+    }
+  );
+
+  const distribution = new cloudfront.Distribution(
+    scope,
+    "CloudFrontDistribution",
+    {
+      certificate: cloudfrontCertificateArn
+        ? acm.Certificate.fromCertificateArn(
+            scope,
+            "certArn",
+            cloudfrontCertificateArn
+          )
+        : undefined,
+      domainNames: cloudfrontDomainName ? [cloudfrontDomainName] : [],
+      defaultBehavior: {
+        origin:
+          cloudfrontOrigins.S3BucketOrigin.withOriginAccessControl(uiBucket),
+        allowedMethods: cloudfront.AllowedMethods.ALLOW_GET_HEAD,
+        viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+        cachePolicy: cloudfront.CachePolicy.CACHING_OPTIMIZED,
+        compress: true,
+        responseHeadersPolicy: securityHeadersPolicy,
+      },
+      defaultRootObject: "index.html",
+      enableLogging: true,
+      logBucket: logBucket,
+      httpVersion: cloudfront.HttpVersion.HTTP2,
+      errorResponses: [
+        {
+          httpStatus: 403,
+          responseHttpStatus: 200,
+          responsePagePath: "/index.html",
+        },
+      ],
+    }
+  );
+  distribution.applyRemovalPolicy(
+    isDev ? RemovalPolicy.DESTROY : RemovalPolicy.RETAIN
+  );
+  if (!isLocalStack) {
+    const waf = setupWaf(scope, stage, project); // vpnIpSetArn, vpnIpv6SetArn
+    distribution.attachWebAclId(waf.webAcl.attrArn);
+  }
+
+  const applicationEndpointUrl = `https://${distribution.distributionDomainName}/`;
+
+  addIamPropertiesToBucketAutoDeleteRole(
+    scope,
+    iamPermissionsBoundary.managedPolicyArn,
+    iamPath
+  );
+
+  return {
+    cloudfrontDistributionId: distribution.distributionId,
+    distribution,
+    applicationEndpointUrl,
+    s3BucketName: uiBucket.bucketName,
+    uiBucket,
+  };
+}
+
+function setupWaf(
+  scope: Construct,
+  stage: string,
+  project: string
+  /*
+   * vpnIpSetArn?: string,
+   * vpnIpv6SetArn?: string,
+   */
+) {
+  return new WafConstruct(
+    scope,
+    "CloudfrontWafConstruct",
+    {
+      name: `${project}-${stage}-ui`,
+      blockByDefault: false,
+    },
+    "CLOUDFRONT"
+  );
+  /*
```

Comment:
This could be turned back on at the drop of a hat, just didn't want to confuse devs in the meantime.